app/src/main/java/com/framgia/music_51/data/resource/TrackLocalDataSource.java

```diff
@@ -0,0 +1,42 @@
+package com.framgia.music_51.data.resource;
+
+import android.arch.lifecycle.LiveData;
+
+import com.framgia.music_51.data.TrackDataSource;
+import com.framgia.music_51.data.model.Track;
+import com.framgia.music_51.data.resource.sql.TrackDao;
+
+import java.util.List;
+
+public class TrackLocalDataSource implements TrackDataSource.Local {
+    private static TrackLocalDataSource sInstance;
+    private TrackDao mDao;
+
+    private TrackLocalDataSource(TrackDao dao) {
+        mDao = dao;
+    }
+
+    public static TrackLocalDataSource getInstance(TrackDao noteDao) {
+        if (sInstance == null) {
+            synchronized (TrackLocalDataSource.class) {
```

Comment:
trường hợp này cần double check

https://en.wikipedia.org/wiki/Double-checked_locking
// Broken multithreaded version
// "Double-Checked Locking" idiom
class Foo {
    private Helper helper;
    public Helper getHelper() {
        if (helper == null) {
            synchronized (this) {
                if (helper == null) {
                    helper = new Helper();
                }
            }
        }
        return helper;
    }

    // other functions and members...
}